Source/Core/Core/AchievementManager.cpp

```diff
@@ -981,8 +1003,8 @@ AchievementManager::ResponseType AchievementManager::ResolveHash(Hash game_hash)
   if (r_type == ResponseType::SUCCESS)
   {
     std::lock_guard lg{m_lock};
-    m_game_id = hash_data.game_id;
-    INFO_LOG_FMT(ACHIEVEMENTS, "Hashed game ID {} for RetroAchievements.", m_game_id);
+    *game_id = hash_data.game_id;
```

Comment:
Locking around a passed pointer like that is really confusing and prone to errors. Just return the `game_id` without the lock and have the caller deal with potential locking.

Reply:
Honestly this looks more like an oversight on my part - the caller already creates a fresh pointer, passes it into this, locks, and then copies the value into its permanent location. Guess I forgot to remove this lock_guard when I did that.